cloud/organization/organization_token.go

```diff
@@ -158,35 +227,356 @@ func getOrganizationToken(id, name, message string, tokens []astrocore.ApiToken)
+		if err != nil {
+			return err
+		}
+		updateOrganizationAPITokenRoles := astrocore.UpdateOrganizationApiTokenRoles{
+			Organization: role,
+			Workspace:    &apiTokenWorkspaceRoles,
+		}
+		UpdateOrganizationAPITokenRequest.Roles = updateOrganizationAPITokenRoles
+	}
+
+	resp, err := client.UpdateOrganizationApiTokenWithResponse(httpContext.Background(), ctx.OrganizationShortName, apiTokenID, UpdateOrganizationAPITokenRequest)
+	if err != nil {
+		return err
+	}
+	err = astrocore.NormalizeAPIError(resp.HTTPResponse, resp.Body)
+	if err != nil {
+		return err
+	}
+	fmt.Fprintf(out, "Astro Organization API token %s was successfully updated\n", token.Name)
+	return nil
+}
+
+// rotate a organization API token
+func RotateToken(id, name string, cleanOutput, force bool, out io.Writer, client astrocore.CoreClient) error {
+	ctx, err := context.GetCurrentContext()
+	if err != nil {
+		return err
+	}
+	if ctx.OrganizationShortName == "" {
+		return user.ErrNoShortName
+	}
+
+	var token astrocore.ApiToken
+
+	if id == "" {
+		tokens, err := getOrganizationTokens(client)
+		if err != nil {
+			return err
+		}
+		token, err = getOrganizationToken(id, name, "\nPlease select the Organization API token you would like to add to the Workspace:", tokens)
+		if err != nil {
+			return err
+		}
+	} else {
+		token, err = getOrganizationTokenByID(id, ctx.OrganizationShortName, client)
+		if err != nil {
+			return err
+		}
+	}
+	apiTokenID := token.Id
+
+	if !force {
+		fmt.Println("WARNING: API Token rotation will invalidate the current token and cannot be undone.")
+		i, _ := input.Confirm(
+			fmt.Sprintf("\nAre you sure you want to rotate the %s API token?", ansi.Bold(token.Name)))
+
+		if !i {
+			fmt.Println("Canceling token rotation")
+			return nil
+		}
+	}
+	resp, err := client.RotateOrganizationApiTokenWithResponse(httpContext.Background(), ctx.OrganizationShortName, apiTokenID)
+	if err != nil {
+		return err
+	}
+	err = astrocore.NormalizeAPIError(resp.HTTPResponse, resp.Body)
+	if err != nil {
+		return err
+	}
+	APIToken := resp.JSON200
+	if cleanOutput {
+		fmt.Println(*APIToken.Token)
+	} else {
+		fmt.Fprintf(out, "\nAstro Organization API token %s was successfully rotated\n", name)
+		fmt.Println("Copy and paste this API token for your records.")
+		fmt.Println("\n" + *APIToken.Token)
+		fmt.Println("\nYou will not be shown this API token value again.")
+	}
+	return nil
+}
+
+// delete a organizations api token
+func DeleteToken(id, name string, force bool, out io.Writer, client astrocore.CoreClient) error {
+	ctx, err := context.GetCurrentContext()
+	if err != nil {
+		return err
+	}
+	if ctx.OrganizationShortName == "" {
+		return user.ErrNoShortName
+	}
+
+	var token astrocore.ApiToken
+
+	if id == "" {
+		tokens, err := getOrganizationTokens(client)
+		if err != nil {
+			return err
+		}
+		token, err = getOrganizationToken(id, name, "\nPlease select the Organization API token you would like to add to the Workspace:", tokens)
```

Comment:
TODO this looks like wrong text likely ill need to fix it in the place it was copied from as well